src/main/java/seedu/taassist/storage/JsonAdaptedModuleClass.java

```diff
@@ -1,38 +1,56 @@
 package seedu.taassist.storage;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
 import com.fasterxml.jackson.annotation.JsonCreator;
-import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonProperty;
 
 import seedu.taassist.commons.exceptions.IllegalValueException;
 import seedu.taassist.model.moduleclass.ModuleClass;
+import seedu.taassist.model.session.Session;
 
 /**
  * Json-friendly version of {@link ModuleClass}.
  */
 class JsonAdaptedModuleClass {
 
     private final String className;
+    private final List<JsonAdaptedSession> sessions = new ArrayList<>();
 
     /**
-     * Constructs a {@code JsonAdaptedModuleClass} with the given {@code className}.
+     * Constructs a {@code JsonAdaptedModuleClass} with the given {@code className} and list
+     * of {@code Session}-s.
      */
     @JsonCreator
-    public JsonAdaptedModuleClass(String className) {
+    public JsonAdaptedModuleClass(@JsonProperty("name") String className,
+                                  @JsonProperty("sessions") List<JsonAdaptedSession> sessions) {
         this.className = className;
+        if (sessions != null) {
+            this.sessions.addAll(sessions);
+        }
     }
 
     /**
      * Converts a given {@code ModuleClass} into this class for Jackson use.
      */
     public JsonAdaptedModuleClass(ModuleClass source) {
         className = source.className;
+        sessions.addAll(source.getSessions().stream().map(JsonAdaptedSession::new).collect(Collectors.toList()));
     }
 
-    @JsonValue
+    @JsonGetter("name")
     public String getClassName() {
         return className;
     }
 
+    @JsonGetter("sessions")
+    public List<JsonAdaptedSession> getSessions() {
+        return sessions;
+    }
+
```

Comment:
I believe this can be done by annotating the declaration above with `@JsonProperty("name")` and such. Try it out. 

Also this annotating may not be required. I think the default name of the variable is used if no annotation is provided. See `JsonAdaptedStudent` in #112. 

Same comment goes for `JsonAdaptedSession`.